File: game_server/logic_server/src/main/java/chr/equip/EquipEntity.java
package chr.equip;

import base.IEntity;
import chr.attr.AttrEntity;
import config.bean.EBattleAttribute;
import config.bean.Equip;
import config.provider.EquipInfoProvider;
import define.EntityType;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by think on 2017/9/14.
 */
public class EquipEntity extends AttrEntity {
	private int equipId;

	private int level;

	private int jinglianLevel;

	private int jinglianExp;

	private int starLevel;

	private int starExp;

	private int starBless;

	private int goldLevel;


	public EquipEntity() {
		super(EntityType.EQUIP);
		level = 1;
	}

	public int getEquipId() {
		return equipId;
	}

	public void setEquipId(int equipId) {
		this.equipId = equipId;
	}

	public int getGoldLevel() {
		return goldLevel;
	}

	public void setGoldLevel(int goldLevel) {
		this.goldLevel = goldLevel;
	}

	public int getStarLevel() {
		return starLevel;
	}

	public void setStarLevel(int starLevel) {
		this.starLevel = starLevel;
	}

	public int getStarExp() {
		return starExp;
	}

	public void setStarExp(int starExp) {
		this.starExp = starExp;
	}

	public int getStarBless() {
		return starBless;
	}

	public void setStarBless(int starBless) {
		this.starBless = starBless;
	}

	public int getLevel() {
		return level;
	}

	public void setLevel(int level) {
		this.level = level;
	}

	public int getJinglianLevel() {
		return jinglianLevel;
	}

	public void setJinglianLevel(int jinglianLevel) {
		this.jinglianLevel = jinglianLevel;
	}

	public int getJinglianExp() {
		return jinglianExp;
	}

	public void setJinglianExp(int jinglianExp) {
		this.jinglianExp = jinglianExp;
	}

	public static EquipEntity createEquip(long entityId, int confId) {
		EquipEntity entity = new EquipEntity();
		Equip conf = EquipInfoProvider.getInst().getEquipConfById(confId);

		return entity;
	}
}

File: godsg/Assets/client_code/logic/entity/EquipEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class EquipEntity
{
    private long equipId;

    private int confId;

    private int level;

    private int jinglianLevel;

    private int jinglianExp;

    private int starLevel;

    private int starExp;

    private int starBless;

    private int goldLevel;

    private Dictionary<EBattleAttribute, int> attributes;

    public Dictionary<EBattleAttribute, int> Attributes
    {
        get
        {
            return this.attributes;
        }
        set
        {
            this.attributes = value;
        }
    }

    public int Level
    {
        get
        {
            return level;
        }

        set
        {
            level = value;
        }
    }

    public int JinglianLevel
    {
        get
        {
            return jinglianLevel;
        }

        set
        {
            jinglianLevel = value;
        }
    }

    public int JinglianExp
    {
        get
        {
            return jinglianExp;
        }

        set
        {
            jinglianExp = value;
        }
    }

    public int StarLevel
    {
        get
        {
            return starLevel;
        }

        set
        {
            starLevel = value;
        }
    }

    public int StarExp
    {
        get
        {
            return starExp;
        }

        set
        {
            starExp = value;
        }
    }

    public int StarBless
    {
        get
        {
            return starBless;
        }

        set
        {
            starBless = value;
        }
    }

    public int GoldLevel
    {
        get
        {
            return goldLevel;
        }

        set
        {
            goldLevel = value;
        }
    }

    public long EquipId
    {
        get
        {
            return equipId;
        }

        set
        {
            equipId = value;
        }
    }

    public int ConfId
    {
        get
        {
            return confId;
        }

        set
        {
            confId = value;
        }
    }
}
